libdrizzle/binlog.cc

```diff
@@ -493,9 +493,9 @@ void drizzle_binlog_get_filename(drizzle_st *con, char **filename, int file_inde
                             row_count - 1 : file_index;
 
     drizzle_row_t row = drizzle_row_index(result, row_index);
-    *filename =( char*)realloc(*filename, strlen(row[0]));
+    *filename = (char*)realloc(*filename, strlen(row[0]) + 1);
     memcpy(*filename, row[0], strlen(row[0]));
-    *filename[strlen(row[0])] = '\0';
+    (*filename)[strlen(row[0])] = '\0';
```

Comment:
I forgot about this inferior language, but @andreas-bok-sociomantic reminded me.
The main point was not to have double call to `strlen`, which is addressed now.